test/test-test.js

```diff
@@ -233,6 +233,63 @@
             }).call({}, "arg1", {}, done);
         },
 
+        "async promise-based test restores the sandbox when the promise is fulfilled": function (done) {
```

Comment:
Prefix this section with a guard clause a la what you can see done [here](https://github.com/joelmukuthu/sinon/blob/2047100d5b26155c9879ac14e2d3c765e8700e57/test/sandbox-test.js#L10) and [here](https://github.com/joelmukuthu/sinon/blob/2047100d5b26155c9879ac14e2d3c765e8700e57/test/sandbox-test.js#L100).